packages/server/graphql/public/mutations/createOAuthAPIProvider.ts

```diff
@@ -0,0 +1,66 @@
+import {GraphQLError} from 'graphql'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import generateUID from '../../../generateUID'
+import {generateOAuthClientId, generateOAuthClientSecret} from '../../../oauth2/credentials'
+import {validateOAuthScopes} from '../../../oauth2/oauthScopes'
+import getKysely from '../../../postgres/getKysely'
+import {selectOAuthAPIProvider} from '../../../postgres/select'
+import type {Oauthscopeenum} from '../../../postgres/types/pg'
+import publish from '../../../utils/publish'
+import type {MutationResolvers} from '../resolverTypes'
+
+const createOAuthAPIProvider: MutationResolvers['createOAuthAPIProvider'] = async (
+  _source,
+  {input},
+  _context
+) => {
+  const {orgId, name, redirectUris, scopes} = input
+
+  const clientId = generateOAuthClientId()
+  const clientSecret = generateOAuthClientSecret()
+
+  if (!validateOAuthScopes(scopes)) {
+    throw new GraphQLError('Invalid scopes. Only graphql:read and graphql:write are allowed.', {
+      extensions: {
+        code: 'BAD_USER_INPUT'
+      }
+    })
+  }
+
+  const pg = getKysely()
+
+  const providerId = generateUID()
```

Comment:
-1 we've deprecated `generateUID`, is it possible to insert this object directly into the DB & get the serial integer ID from the response? GraphQL can make it opaque for the client, if needed (see `CipherId.toClient`)